app/controllers/api_controller/parser.rb

```diff
@@ -133,6 +133,14 @@ def parse_tenant(resource)
       parse_id(resource, :tenants) unless resource.blank?
     end
 
+    def parse_provider(resource)
+      parse_id(resource, :providers) unless resource.blank?
```

Comment:
Is there a way we could ensure this method isn't called if `resource` is blank?. We know that it can't be `nil` because of https://github.com/ManageIQ/manageiq/pull/9858/files#diff-114486ea47c458e2efe222564ab4c32bR178....perhaps you could do some extra validation in `validate_profile_data` to ensure that the incoming data is properly formed?